### PR TITLE
Dev

### PR DESCRIPTION
**Pull request by Yuriy27**

### CODE REVIEW

File: softgroup/basis/common-router/src/main/java/com/softgroup/common/router/api/AbstractRequestHandler.java [hunk 1]
```diff
@@ -1,20 +1,30 @@
...
 
+import java.util.Map;
+
 public abstract class AbstractRequestHandler<T extends RequestData, R extends ResponseData> implements RequestHandler {
-	@Override
-	public String getName() {
-		return null;
-	}
+
+	public abstract String getName();
```
Comment (Dominys) [inline softgroup/basis/common-router/src/main/java/com/softgroup/common/router/api/AbstractRequestHandler.java]: @Yuriy27 No need to make this abstract. It's already defined in interface.

File: softgroup/basis/common-router/src/main/java/com/softgroup/common/router/api/AbstractRequestHandler.java [hunk 2]
```diff
@@ -1,20 +1,30 @@
...
+
+	public abstract Response<R> doHandle(Request<T> msg);
 
 	@Override
 	public Response<R> handle(Request<?> msg) {
-		return null;
+		JacksonDataMapper mapper = new JacksonDataMapper();
+		Map<String, Object> map = mapper.convertToMap(msg);
+		RequestData data = mapper.convert(map, RequestData.class);
```
Comment (Dominys) [inline softgroup/basis/common-router/src/main/java/com/softgroup/common/router/api/AbstractRequestHandler.java]: @Yuriy27 no need to make two conversions. Next - do this work? Test this.

File: softgroup/basis/common-router/src/main/java/com/softgroup/common/router/api/AbstractRequestHandler.java [hunk 3]
```diff
@@ -1,20 +1,30 @@
...
-	}
+
+	public abstract String getName();
+
+	public abstract Response<R> doHandle(Request<T> msg);
 
 	@Override
 	public Response<R> handle(Request<?> msg) {
-		return null;
+		JacksonDataMapper mapper = new JacksonDataMapper();
```
Comment (Dominys) [inline softgroup/basis/common-router/src/main/java/com/softgroup/common/router/api/AbstractRequestHandler.java]: @Yuriy27 Why we need to create new mapper ?

File: softgroup/basis/common-router/src/main/java/com/softgroup/common/router/impl/MainRouter.java [hunk 4]
```diff
@@ -0,0 +1,19 @@
...
+import com.softgroup.common.router.api.AbstractRouterHandler;
+import com.softgroup.common.router.api.CommonRouterHandler;
+import com.softgroup.common.router.api.RouterHandler;
+import org.springframework.stereotype.Component;
+
+/**
+ * Created by yuriy on 28.02.17.
+ */
+@Component
+public class MainRouter
```
Comment (Dominys) [inline softgroup/basis/common-router/src/main/java/com/softgroup/common/router/impl/MainRouter.java]: @Yuriy27 Lets rename this to FirstRouter. It is not main.

File: softgroup/basis/common-router/src/main/java/com/softgroup/common/router/impl/MainRouter.java [hunk 5]
```diff
@@ -0,0 +1,19 @@
...
+        extends AbstractRouterHandler<RouterHandlerFactory<CommonRouterHandler>>
```
Comment (Dominys) [inline softgroup/basis/common-router/src/main/java/com/softgroup/common/router/impl/MainRouter.java]: @Yuriy27 RouterHandlerFactory is implementation. We should not depend on implementation.

File: softgroup/services/authorization/authorization-imp/src/main/java/com/softgroup/authorization/impl/LoginRequestHandler.java [hunk 6]
```diff
@@ -0,0 +1,33 @@
...
+public class LoginRequestHandler
+        extends AbstractRequestHandler<LoginRequest, LoginResponse>
+        implements AuthorizationRequestHandler {
+
+    @Override
+    public String getName() {
+        return "login";
+    }
+
+    public Response<LoginResponse> doHandle(Request<LoginRequest> msg) {
```
Comment (Dominys) [inline softgroup/services/authorization/authorization-imp/src/main/java/com/softgroup/authorization/impl/LoginRequestHandler.java:24]: @Yuriy27 Nice temporary plug!

**Comment by Dominys** [comment]
@Yuriy27 General note - use proper branch names. Dev is not provide any information.

File: softgroup/basis/common-router/src/main/java/com/softgroup/common/router/api/AbstractRequestHandler.java [hunk 7]
```diff
@@ -1,20 +1,33 @@
...
 	public String getName() {
 		return null;
 	}
 
+	public abstract Response<R> doHandle(Request<T> msg);
+
 	@Override
 	public Response<R> handle(Request<?> msg) {
-		return null;
+		Map<String, Object> map = mapper.convertToMap(msg);
```
Comment (Yuriy27) [inline softgroup/basis/common-router/src/main/java/com/softgroup/common/router/api/AbstractRequestHandler.java:25]: I dont know how to convert Request<?> to Request<T> using DataMapper one time. I need a hint.

File: softgroup/basis/common-router/src/main/java/com/softgroup/common/router/api/AbstractRequestHandler.java [hunk 8]
```diff
@@ -12,13 +12,16 @@
 
 public abstract class AbstractRequestHandler<T extends RequestData, R extends ResponseData> implements RequestHandler {
 
-	public abstract String getName();
+	DataMapper mapper = new JacksonDataMapper();
```
Comment (Dominys) [inline softgroup/basis/common-router/src/main/java/com/softgroup/common/router/api/AbstractRequestHandler.java]: @Yuriy27 Why new mapper ?

File: softgroup/basis/common-router/src/main/java/com/softgroup/common/router/api/AbstractRequestHandler.java [hunk 9]
```diff
@@ -12,13 +12,16 @@
...
+
+	public String getName() {
+		return null;
```
Comment (Dominys) [inline softgroup/basis/common-router/src/main/java/com/softgroup/common/router/api/AbstractRequestHandler.java]: @Yuriy27 Why getName returns null ?

File: softgroup/basis/common-router/src/main/java/com/softgroup/common/router/api/AbstractRequestHandler.java (hunk 7 above)
Comment (Dominys) [reply softgroup/basis/common-router/src/main/java/com/softgroup/common/router/api/AbstractRequestHandler.java:25]: @Yuriy27 For example add new method to dataMapper

File: softgroup/basis/common-router/src/main/java/com/softgroup/common/router/api/AbstractRequestHandler.java [hunk 10]
```diff
@@ -1,20 +1,33 @@
...
 import com.softgroup.common.protocol.RequestData;
 import com.softgroup.common.protocol.Response;
 import com.softgroup.common.protocol.ResponseData;
 
+import java.util.Map;
+
 public abstract class AbstractRequestHandler<T extends RequestData, R extends ResponseData> implements RequestHandler {
-	@Override
+
+	DataMapper mapper = new JacksonDataMapper();
```
Comment (Dominys) [inline softgroup/basis/common-router/src/main/java/com/softgroup/common/router/api/AbstractRequestHandler.java:15]: @Yuriy27 Why new mapper ???

File: softgroup/basis/common-router/src/main/java/com/softgroup/common/router/api/AbstractRequestHandler.java [hunk 11]
```diff
@@ -1,20 +1,33 @@
...
+
 	public String getName() {
 		return null;
```
Comment (Dominys) [inline softgroup/basis/common-router/src/main/java/com/softgroup/common/router/api/AbstractRequestHandler.java:18]: @Yuriy27 Why getName returns null ?

File: softgroup/basis/common-router/src/main/java/com/softgroup/common/router/api/AbstractRequestHandler.java (hunk 7 above)
Comment (Dominys) [reply softgroup/basis/common-router/src/main/java/com/softgroup/common/router/api/AbstractRequestHandler.java:25]: @Yuriy27 Make test for this conversion!

**Pull request closed without merging**